plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/simple/SimpleProcessor.java

```diff
@@ -55,10 +55,10 @@ public void execute(Data data) {
     public static class Factory implements Processor.Factory {
 
         public Processor create(Map<String, Object> config) {
-            String path = (String) config.get("path");
-            String expectedValue = (String) config.get("expected_value");
-            String addField = (String) config.get("add_field");
-            String addFieldValue = (String) config.get("add_field_value");
+            String path = (String) config.remove("path");
+            String expectedValue = (String) config.remove("expected_value");
+            String addField = (String) config.remove("add_field");
+            String addFieldValue = (String) config.remove("add_field_value");
```

Comment:
I think it makes sense to do it here too, especially the need for remove especially should be contained in the utils method I think.